configuration/pih/htmlforms/hiv/section-hiv-state.xml

```diff
@@ -156,10 +156,248 @@
+            startDate: "",
+            date: null,
+            encounterDisplay: ""
+        };
+
+        
+           jq.getJSON(apiBaseUrl + "/obs", {
+            concept: inhStartdateConceptUuid,
+            patient: patientUuid,
+            v: 'custom:(uuid,display,obsDatetime,value,encounter:(display,encounterDatetime,location:(display),encounterType:(display)))'
+          },
+          function( data ){
+            if (data.results.length > 0) {
+                for (let index = 0; index &lt; data.results.length; ++index) {
+                    let inhStartDateObs = data.results[index];
+                    let inhStartDateObsDate = new Date(inhStartDateObs.obsDatetime);
+                    let tmpInhStartDateObs = {
+                      startDate: "",
+                      date: null,
+                      encounterDisplay: ""
+                    };
+                   
+
+                    if( (currentEncounterDate.getTime() >=  inhStartDateObsDate.getTime()) ||
+                        (currentEncounterDate.toDateString() === inhStartDateObsDate.toDateString())) {
+                            if (typeof inhStartDateObs.value !== "undefined" &amp;&amp; inhStartDateObs.value !== null &amp;&amp; (inhStartDateObs.value.length &gt; 0)) {
+                                tmpInhStartDateObs.startDate = inhStartDateObs.value;
+                                tmpInhStartDateObs.date = inhStartDateObsDate;
+                                tmpInhStartDateObs.encounterDisplay = inhStartDateObs.encounter.display;
+
+                                
+                              }
+                    }
+                    if ( (tmpInhStartDateObs.date != null) &amp;&amp;
+                        ((lastRecordedStartDate.date == null) || (lastRecordedStartDate.date.getTime() &lt; tmpInhStartDateObs.date.getTime()) ) ) {
+
+                        lastRecordedStartDate.startDate = tmpInhStartDateObs.startDate;
+                        lastRecordedStartDate.date = tmpInhStartDateObs.date;
+                        lastRecordedStartDate.encounterDisplay = tmpInhStartDateObs.encounterDisplay;
+                    }
+                }
+            }
+            if (lastRecordedStartDate.date != null) {
+                  let domEl = getField('inhStartDate.value')
+                   domEl.datepicker("setDate", new Date(lastRecordedStartDate.startDate));
+                  }
+         
+          });
+    });
+</script>
+</excludeIf>
+
+
+<!-- This method allows to recover the type of izoniazide, if it is primary or secondary -->
+<excludeIf velocityTest="$fn.getObs($encounter, 'PIH:13786')">
+<script type="text/javascript">
+
+    jq(document).ready(function() {
+        
+        const contextPath = window.location.href.split('/')[3];
+        const apiBaseUrl =  "/" + contextPath + "/ws/rest/v1";
+        const patientUuid = '<lookup expression="patient.uuid"/>';
+
+        let currentEncounterDate = new Date();
```

Comment:
Same comment.  There are some utility methods I started adding to help with things like this and to have this defined in one place rather than to have to re-invent/copy-paste everywhere.  See:  https://github.com/PIH/openmrs-config-pihemr/blob/master/configuration/pih/scripts/global/htmlFormFunctions.js